presto-main/src/test/java/com/facebook/presto/type/TestIpAddressOperators.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import com.facebook.presto.spi.type.SqlIpAddress;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.IpAddressType.IPADDRESS;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+
+public class TestIpAddressOperators
+    extends AbstractTestFunctions
+{
+    @Test
+    public void testVarcharToIpAddressCast()
+    {
+        assertFunction("CAST('::ffff:1.2.3.4' AS IPADDRESS)", IPADDRESS, new SqlIpAddress("1.2.3.4"));
+        assertFunction("CAST('1.2.3.4' AS IPADDRESS)", IPADDRESS, new SqlIpAddress("::ffff:1.2.3.4"));
+        assertFunction("CAST('1.2.3.4' AS IPADDRESS)", IPADDRESS, new SqlIpAddress("::ffff:102:304"));
+        assertFunction("CAST('2001:0db8:0000:0000:0000:ff00:0042:8329' AS IPADDRESS)", IPADDRESS, new SqlIpAddress("2001:db8::ff00:42:8329"));
+        assertFunction("CAST('2001:db8::ff00:42:8329' AS IPADDRESS)", IPADDRESS, new SqlIpAddress("2001:db8:0:0:0:ff00:42:8329"));
+    }
+
+    @Test
+    public void testIpAddressToVarcharCast()
+    {
+        assertFunction("CAST(IPADDRESS'::ffff:1.2.3.4' AS VARCHAR)", VARCHAR, "1.2.3.4");
+        assertFunction("CAST(IPADDRESS'::ffff:102:304' AS VARCHAR)", VARCHAR, "1.2.3.4");
+        assertFunction("CAST(IPADDRESS'2001:0db8:0000:0000:0000:ff00:0042:8329' AS VARCHAR)", VARCHAR, "2001:db8:0:0:0:ff00:42:8329");
+        assertFunction("CAST(IPADDRESS'2001:db8::ff00:42:8329' AS VARCHAR)", VARCHAR, "2001:db8:0:0:0:ff00:42:8329");
```

Comment:
Hmm. InetAddress::toString() returns this form. If we want the that we will need to do the conversion ourselves? Didn't see a function returning that form.